UI/goliveapi-postdata.cpp

```diff
@@ -0,0 +1,50 @@
+#include "goliveapi-postdata.hpp"
+
+#include <nlohmann/json.hpp>
+
+#include "system-info.hpp"
+
+nlohmann::json
+constructGoLivePost(QString streamKey,
+		    const std::optional<uint64_t> &maximum_aggregate_bitrate,
+		    const std::optional<uint32_t> &maximum_video_tracks,
+		    bool vod_track_enabled)
+{
+	using json = nlohmann::json;
+
+	json post_data;
+	post_data["service"] = "IVS";
+	post_data["schema_version"] = "2023-05-10";
+	post_data["authentication"] = streamKey.toStdString();
+
+	auto &capabilities = post_data["capabilities"];
+	capabilities["gpu"] = system_gpu_data();
+	capabilities.update(system_info(), true);
```

Comment:
The user must be able to reject having their system information being sent to a third party.

Reply:
> The user must be able to reject having their system information being sent to a third party.

The user can reject having their system information being sent by not enabling Enhanced Broadcasting or Multitrack Video in the UI.

In other words: this information is never sent to the destination service if the "Enable Enhanced Broadcasting" or "Enable Multitrack Video" are unchecked. In the UI, the user is presented with a checkbox to enable Enhanced Broadcasting or Multitrack Video.

If the feature is enabled via the UI checkbox, then `GetClientCapabilities` is required to fetch the configuration settings and further, the `GetClientCapabilities` API requires this information to vend the best possible, safe ladder for every setup.

Reply:
This is bad, the user needs be aware of this and make an explicit choice to send their data.
Hiding this detail is really bad, IIRC the data that are sent can be assigned as PII.

Reply:
The user is made aware of this with an explicit opt-in and notification when they enable the feature. Nothing is hidden, and I'm not sure why you're assuming it is.

Reply:
I didn't see the notification part when the check box is toggled in settings, maybe I'm looking at the wrong place.

Sorry then.

Reply:
Found it, it's a label. I'm sorry.

To me, it looks like a bad pattern to block a feature based on if the user wants to share data or not.
But this is out of scope.

This can be put as resolved.